### PR TITLE
React: Standardize error handling 

### LINKED ISSUE #736: React: Consolidate HTTP error alerts
Currently there are a variety of error-related snackbars in the front-end. We should institute a single handler (maybe a hook that wraps `useSnackbar`?) for display to the user. It should print the code and message for the various errors the user might see, including 400, 404, 406, 415, 422, and 500.

### PR DESCRIPTION
**Pull request by hannahle**
Closes #736 

- Created useErrorSnackbar hook as a wrapper for useSnackbar to standardize error handling 

**To-do:**
- If this looks good, I'll go ahead and refactor the current enqueueSnackbar/ useSnackbar instances on our front-end.

### CODE REVIEW

File: react/src/SearchVariants/components/SearchVariantsPage.tsx [hunk 1]
```diff
@@ -124,7 +123,7 @@ const SearchVariantsPage: React.FC<SearchVariantsPageProps> = () => {
                         </Grid>
                         <Grid item>
                             <Button
-                                disabled={disableControls}
+                                // disabled={disableControls}
```
Comment (frewmack) [inline react/src/SearchVariants/components/SearchVariantsPage.tsx:126]: This should stay un-commented. We don't want the user to have the option to download a report if they haven't specified what genes they want in their report.

**Review by frewmack — CHANGES_REQUESTED** [review]
Looks great! Minor request unrelated to the snackbar wrapper and we're good to go.

File: react/src/SearchVariants/components/SearchVariantsPage.tsx (hunk 1 above)
Comment (hannahle) [reply react/src/SearchVariants/components/SearchVariantsPage.tsx:126]: That makes sense! I'll uncomment - was leaving it there as a way to test the appearance of the snackbar.

**Review by frewmack — APPROVED** [review]
LGTM!

**Review by cklamann — APPROVED** [review]
Looks good to me!

**Review by frewmack — APPROVED** [review]
Looks good to me!

**Pull request merged**